backend-test/Controllers/CourseControllerTests.hs

```diff
@@ -6,17 +6,88 @@ Module that contains the tests for the functions in the Course Controller module
+        ("description", "An introduction to statistical inference and practice. Statistical models and parameters, estimators of parameters and their statistical properties, methods of estimation, confidence intervals, hypothesis testing, likelihood function, the linear model. Use of statistical computation for data analysis and simulation."),
+        ("prereqs", "STA237H1/  STA247H1/  STA257H1/  STAB52H3/  STA256H5"),
+        ("exclusions", "ECO220Y1/  ECO227Y1/  GGR270H1/  PSY201H1/  SOC300H1/  SOC202H1/  SOC252H1/  STA220H1/  STA221H1/  STA255H1/  STA248H1/  STA261H1/  STA288H1/  EEB225H1/  STAB22H3/  STAB27H3/  STAB57H3/  STA220H5/  STA221H5/  STA258H5/  STA260H5/  ECO220Y5/  ECO227Y5"),
+        ("breadth", "The Physical and Mathematical Universes (5)"), 
+        ("distribution", "null"),
+        ("prereqString", "STA237H1/  STA247H1/  STA257H1/  STAB52H3/  STA256H5"),
+        ("coreqs", "CSC108H1/  CSC110Y1/  CSC148H1 *Note: the corequisite may be completed either concurrently or in advance."),
+        ("videoUrls", "[\"https://example.com/video1\", \"https://example.com/video2\"]")
+        ],
+        "{\"allMeetingTimes\":[],\"breadth\":null,\"coreqs\":\"CSC108H1/  CSC110Y1/  CSC148H1 *Note: the corequisite may be completed either concurrently or in advance.\",\"description\":\"An introduction to statistical inference and practice. Statistical models and parameters, estimators of parameters and their statistical properties, methods of estimation, confidence intervals, hypothesis testing, likelihood function, the linear model. Use of statistical computation for data analysis and simulation.\",\"distribution\":null,\"exclusions\":\"ECO220Y1/  ECO227Y1/  GGR270H1/  PSY201H1/  SOC300H1/  SOC202H1/  SOC252H1/  STA220H1/  STA221H1/  STA255H1/  STA248H1/  STA261H1/  STA288H1/  EEB225H1/  STAB22H3/  STAB27H3/  STAB57H3/  STA220H5/  STA221H5/  STA258H5/  STA260H5/  ECO220Y5/  ECO227Y5\",\"name\":\"STA238H1\",\"prereqString\":\"STA237H1/  STA247H1/  STA257H1/  STAB52H3/  STA256H5\",\"title\":\"Probability, Statistics and Data Analysis II\",\"videoUrls\":[\"[\\\"https://example.com/video1\\\", \\\"https://example.com/video2\\\"]\"]}"
+    ),
+    
+    ("Course does not exist", 
+       "STA238", 
+       Map.empty, 
+       "null"
+    ),
+    
+    ("No course provided", 
+       "", 
+       Map.empty, 
+       "null"
+    )
+    ]
+
+-- | Run a test case (case, input, expected output) on the retrieveCourse function.
+runRetrieveCourseTest :: String -> T.Text -> Map.Map String String -> String -> Test
+runRetrieveCourseTest label courseName courseData expected =
+    TestLabel label $ TestCase $ do
+        let currCourseName = T.pack $ fromMaybe "" $ Map.lookup "name" courseData
+
+        let courseToInsert = 
+                [ Courses
+                    { coursesCode = currCourseName
+                    , coursesTitle = fmap T.pack $ Map.lookup "title" courseData <|> Nothing
```

Comment:
The `<|> Nothing` isn't necessary, `Map.lookup` will already return `Nothing` if the key isn't found